### PR TITLE
Add beacon for android devices

### PR DESCRIPTION
**Pull request by opdude**
Detect when a device state changes or battery level gets low

### CODE REVIEW

File: salt/beacons/adb.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+def validate(config):
+    '''
+    Validate the beacon configuration
+    '''
+    # Configuration for adb beacon should be a dictionary with states array
+    if not isinstance(config, dict):
+        log.info('Configuration for adb beacon must be a dict.')
+        return False
+    elif "states" not in config.keys():
```
Comment (cachedout) [inline salt/beacons/adb.py]: We prefer single-quotes here, please.

File: salt/beacons/adb.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          adb:
+            - states:
+                - offline
+                - unauthorized
+                - missing
+            - no_devices_event: True
+            - battery_low: 25
+
+    '''
+    global last_state_no_devices
```
Comment (cachedout) [inline salt/beacons/adb.py]: We typically discourage the use of globals. Is there a way to refactor this code so that this is not necessary?

**Comment by cachedout** [comment]
Hi @opdude This looks pretty great. There are some small, mostly style issues such as the use of double-quotes throughout this PR that we would like to see cleaned up. Could you please take a look and let me know what you think?

File: salt/beacons/adb.py (hunk 2 above)
Comment (opdude) [inline salt/beacons/adb.py]: We could save the state to a file if that's preferred?

File: salt/beacons/adb.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+'''
+Beacon to emit adb device state changes for Android devices
+'''
```
Comment (jfindlay) [inline salt/beacons/adb.py]: @opdude, `.. versionadded:: Boron`.

**Comment by cachedout** [comment]
Looks great! Thank you, @opdude

**Pull request merged**